### PR TITLE
task/WG-379: upgrade python to 3.11; reorg Dockerfiles; reorg `pyproject.toml` (first pass)

### PR DESCRIPTION
**Pull request by rstijerina**
## Overview: ##

- Upgrades python to 3.11
- Upgrades all packages
- Reorganizes python dependencies into groups
- Adds a new makefile build target: `make build-dev`
  - Target for local development, production builds will not include dev dependencies

## Related Jira tickets: ##

* [WG-379](https://tacc-main.atlassian.net/browse/WG-379)

### CODE REVIEW

File: geoapi/services/point_cloud.py [hunk 1]
```diff
@@ -31,7 +34,7 @@ def get(database_session, pointCloudId: int) -> PointCloud:
         :param pointCloudId: int
         :return: PointCloud
         """
-        point_cloud = database_session.query(PointCloud).get(pointCloudId)
+        point_cloud = database_session.get(PointCloud, pointCloudId)
```
Comment (rstijerina) [inline geoapi/services/point_cloud.py:37]: https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#orm-query-get-method-moves-to-session

File: geoapi/tests/api_tests/test_vectors_service.py [hunk 2]
```diff
@@ -1,17 +1,31 @@
...
+        "iso_a3": "CHL",
+        "name": "Chile",
+        "pop_est": 17789267,
+    }
 
 
 def test_process_shapefile_missing_additional_files(shapefile_fixture):
-    with pytest.raises(fiona.errors.DriverError):
-        _, _ = next(VectorService.process_shapefile(shapefile_fixture, additional_files=[]))
+    with pytest.raises(pyogrio.errors.DataSourceError):
```
Comment (rstijerina) [inline geoapi/tests/api_tests/test_vectors_service.py:28]: https://geopandas.org/en/latest/docs/user_guide/fiona_to_pyogrio.html

File: geoapi/utils/lidar.py [hunk 3]
```diff
@@ -62,13 +69,22 @@ def get_bounding_box_2d(filePaths: List[str]) -> MultiPolygon:
...
-        min_point = _transform_to_geojson(proj4=proj4, point=tuple(las_file.header.min[:3]))
-        max_point = _transform_to_geojson(proj4=proj4, point=tuple(las_file.header.max[:3]))
-        las_file.close()
+        las_file = laspy.read(input_file)
+        min_point = _transform_to_geojson(
+            proj4=proj4, point=tuple(las_file.header.mins[:3])
+        )
+        max_point = _transform_to_geojson(
+            proj4=proj4, point=tuple(las_file.header.maxs[:3])
+        )
```
Comment (rstijerina) [inline geoapi/utils/lidar.py:78]: https://laspy.readthedocs.io/en/latest/migration.html

File: Makefile [hunk 4]
```diff
@@ -53,3 +53,10 @@ deploy-geoapi:
 .PHONY: deploy-workers
 deploy-workers:
 	docker push $(GEOAPI_WORKERS):$(TAG)
+
+.PHONY: build-dev
```
Comment (nathanfranklin) [inline Makefile:57]: Should we update the README :  `make build` -> `make build-dev`

File: devops/docker-compose.local.yml [hunk 5]
```diff
@@ -9,16 +9,21 @@ networks:
 services:
 
   rabbitmq:
-    image: rabbitmq:3.12.6
+    image: rabbitmq:3-management
```
Comment (nathanfranklin) [inline devops/docker-compose.local.yml:12]: Should we expose the port to use the UI.
```suggestion
    image: rabbitmq:3-management
    ports:
      - 15672:15672
```

File: devops/docker-compose.local.yml [hunk 6]
```diff
@@ -9,16 +9,21 @@ networks:
...
       - RABBITMQ_DEFAULT_VHOST=dev
+    healthcheck:
+      test: rabbitmq-diagnostics -q ping
+      interval: 5s
+      timeout: 30s
+      retries: 5
 
   postgres:
-    image: mdillon/postgis:11-alpine
+    image: postgis/postgis:11-3.0-alpine
```
Comment (nathanfranklin) [inline devops/docker-compose.local.yml:26]:
```suggestion
    image: postgis/postgis:11-3.3-alpine
```

nice.  i see we have moved away from mdillon/postgis on the [production server.](https://github.com/TACC-Cloud/geoapi/blob/main/devops/database/docker-compose-database.yml#L3)

File: devops/pyproject.toml [hunk 7]
```diff
@@ -2,44 +2,59 @@
...
 version = "2.5"
 description = ""
-authors = ["Your Name <you@example.com>"]
+authors = ["DesignSafe-CI <designsafe-ci@tacc.utexas.edu>"]
+package-mode = false
 
 [tool.poetry.dependencies]
-python = "^3.9"
-celery = "5.2.3"
+python = "^3.11"
```
Comment (nathanfranklin) [inline devops/pyproject.toml:9]: 💯

File: devops/docker-compose.local.yml (hunk 6 above)
Comment (rstijerina) [reply devops/docker-compose.local.yml:26]: Yes I meant to confirm - good change to move to official postgis repo in your opinion?

File: geoapi/utils/lidar.py [hunk 8]
```diff
@@ -69,14 +69,13 @@ def get_bounding_box_2d(filePaths: List[str]) -> MultiPolygon:
     for input_file in filePaths:
         proj4 = getProj4(input_file)
 
-        las_file = laspy.file.File(input_file, mode="r-")
+        las_file = laspy.read(input_file)
```
Comment (nathanfranklin) [inline geoapi/utils/lidar.py:72]: need to add `lazrs` python package.  looks like in the new laspy a backed needs to be installed.  i installed lazrs and all works well with that locally.  I'm investigating why the unit tests didn't pick that up (maybe the files we use to test this function don't trigger the error: "LaspyException: No LazBackend selected, cannot decompress data" ).

File: devops/docker-compose.local.yml (hunk 6 above)
Comment (nathanfranklin) [reply devops/docker-compose.local.yml:28]: and deployed we got `postgis/postgis:11-3.3-alpine` insetad of the 3.0.  so we can match that here too.

File: geoapi/utils/lidar.py (hunk 8 above)
Comment (nathanfranklin) [reply geoapi/utils/lidar.py:72]: okay, i see the issue and added a unit test. we were never testing any compressed files (ie. laz). we had some files that had .laz ending but they were uncompressed.  i got a PR targeting this branch and adding a unit test.  see https://github.com/TACC-Cloud/geoapi/pull/224

**Pull request merged**